### PR TITLE
Don't add win_evt logger when on nano.

### PR DESCRIPTION
**Pull request by mcquin**
@chef/client-windows

### CODE REVIEW

File: lib/chef/platform/query_helpers.rb [hunk 1]
```diff
@@ -36,6 +36,28 @@ def windows_server_2003?
...
+      def windows_nano_server?
+        return false unless windows?
+        require 'win32/registry'
+
+        # This method may be called before ohai runs (e.g., it may be used to
+        # determine settings in config.rb). Chef::Win32::Registry.new uses
+        # node attributes to verify the machine architecture which aren't
+        # accessible before ohai runs.
+        nano = nil
+        key = "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Server\\ServerLevels"
```
Comment (ksubrama) [inline lib/chef/platform/query_helpers.rb]: Wont this be true on all server versions of windows?  From my server 2012 workstation:

```
get-itemproperty "hklm:\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Server\\ServerLevels"


ServerCoreExtended : 1
ServerCore         : 1
Server-Gui-Mgmt    : 1
Server-Gui-Shell   : 1
PSPath             : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Server\ServerLevels
PSParentPath       : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Server
PSChildName        : ServerLevels
PSDrive            : HKLM
PSProvider         : Microsoft.PowerShell.Core\Registry
```

Comment (mcquin) [inline lib/chef/platform/query_helpers.rb]: I don't understand the question. What is "this" referring to?

**Comment by mcquin** [comment]
These appveyor failures are unrelated to this change. Additionally, I am unable to reproduce them on this branch and off of master with 32-bit and 64-bit ruby 2.0 on Win2012r2.

File: lib/chef/platform/query_helpers.rb (hunk 1 above)
Comment (ksubrama) [inline lib/chef/platform/query_helpers.rb]: windows_nano_server?  I believe it returns true on any server version of windows (2008, 2012, 2012 server core, 2016 etc.)

Comment (mcquin) [inline lib/chef/platform/query_helpers.rb]: We check for a `"NanoServer"` key with value 1 within `HKLM:\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Server\\ServerLevels"`: https://github.com/chef/chef/blob/mcquin/detect-nano/lib/chef/platform/query_helpers.rb#L51-L53

Comment (mcquin) [inline lib/chef/platform/query_helpers.rb]: On my win2012r2 box:

```
> pry
[1] pry(main)> require_relative "chef-config/lib/chef-config/windows"
=> true
[2] pry(main)> require_relative "lib/chef/platform/query_helpers"
=> true
[3] pry(main)> Chef::Platform.windows_nano_server?
=> false
```

**Comment by chefsalim** [comment]
LGTM :+1:

File: lib/chef/platform/query_helpers.rb (hunk 1 above)
Comment (ksubrama) [inline lib/chef/platform/query_helpers.rb]: *facepalm*  I'm silly and misread the actual key you were reading out.  This makes perfect sense.

**Comment by ksubrama** [comment]
:+1:

**Comment by mcquin** [comment]
The failures were related to the change, after all. We found that removing the entire `Win32` namespace before each spec negatively affected other specs with libraries relying on the win32 library (in this case, rubygems). Though it hasn't caused problems yet, we may want to clean up [this spec](https://github.com/chef/chef/blob/master/spec/unit/win32/registry_spec.rb#L37).

**Comment by chefsalim** [comment]
Hmm, interesting.. I'm stealing some of your spec code for my msi lazy load tests as well.

File: spec/unit/platform/query_helpers_spec.rb [hunk 2]
```diff
@@ -31,7 +31,90 @@
...
+    Win32::Registry::HKEY_LOCAL_MACHINE = hive
+    Win32::Registry::KEY_QUERY_VALUE = key_query_value
+  end
+
+  after do
+    Win32::Registry.send(:remove_const, 'HKEY_LOCAL_MACHINE') if defined?(Win32::Registry::HKEY_LOCAL_MACHINE)
+    Win32::Registry.send(:remove_const, 'KEY_QUERY_VALUE') if defined?(Win32::Registry::KEY_QUERY_VALUE)
+  end
+
+  after(:all) do
```
Comment (ksubrama) [inline spec/unit/platform/query_helpers_spec.rb]: Yay!  I'm glad this works.  Totally stealing it for the other specs.

**Comment by ksubrama** [comment]
:+1:

**Pull request merged**